### PR TITLE
Transcripts: Search UI

### PR DESCRIPTION
**Pull request by leandroalonso**
| 📘 Part of: #1848 | 
|:---:|

**Important**: search logic is not included in this PR, just the UI.

## To test

Ensure the `transcript` feature flag is enabled.

### Showing accessory view

1. Run the app
2. Play any episode with transcripts
3. Tap the transcript button on the shelf
4. Tap "Search"
5. ✅ Keyboard should appear alongside the search view
6. Tap "Search"
7. ✅ The keyboard should disappear but the search view should remain
8. Tap "Done" on the search view
9. ✅ Search view is dismissed

### Closing screen

1. Tap "Search" again
2. Tap "X" to dismiss the screen
3. ✅ The keyboard and the search view should disappear

### Accessibility

1. Go to Settings > Acessibility
2. Increase the font
3. ✅ Open the search and check that all elements are now bigger
4. Re-run the app with RtL pseudo language enabled
5. Check that the search has a clear button and label on the other side

Ps.: there's a bug with how the player is displayed. [I opened an issue here](https://github.com/Automattic/pocket-casts-ios/issues/1952).

### Label (number of results)

1. Open the search again
2. Type something
3. ✅ The clear button should appear and the label "1/10" should move a bit
4. Tap the clear button
5. ✅ The label should move back
6. Type again and then tap "search" on the keyboard
7. ✅ Clear button should disappear and the label will move a bit to cover the space

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/TranscriptSearchAccessoryView.swift [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    lazy var textField: CustomTextField = {
+        let textField = CustomTextField()
+        textField.returnKeyType = .search
+        textField.autocorrectionType = .no
+        textField.spellCheckingType = .no
+        textField.clearButtonMode = .whileEditing
+        textField.layer.cornerRadius = 8
+        textField.backgroundColor = .systemGray3
+
+        textField.rightLabel.text = "1/10"
```
Comment (leandroalonso) [inline podcasts/TranscriptSearchAccessoryView.swift]: Just a placeholder.

File: podcasts.xcodeproj/xcshareddata/xcschemes/pocketcasts.xcscheme [hunk 2]
```diff
@@ -86,6 +86,9 @@
             isEnabled = "YES">
          </CommandLineArgument>
       </CommandLineArguments>
+      <StoreKitConfigurationFileReference
+         identifier = "../Pocket Casts Configuration.storekit">
```
Comment (SergioEstevao) [inline podcasts.xcodeproj/xcshareddata/xcschemes/pocketcasts.xcscheme]: Was this change intended?

File: podcasts/TranscriptSearchAccessoryView.swift [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        title: L10n.done,
+        action: #selector(done),
+        titleColor: .label,
+        highlightedTitleColor: .systemGray,
+        contentInsets: .init(top: 8, leading: 16, bottom: 8, trailing: 16)
+    )
+
+    private lazy var downButton: UIButton = createSymbolButton(
+        imageName: "chevron.down",
+        action: #selector(done),
```
Comment (SergioEstevao) [inline podcasts/TranscriptSearchAccessoryView.swift:46]: I know that the search logic is not done yet, but shouldn't this trigger a different delegate method, like a Next/Previous call?

File: Modules/Utils/Sources/PocketCastsUtils/Extensions/UI/CustomTextView.swift [hunk 4]
```diff
@@ -0,0 +1,77 @@
+#if !os(watchOS)
+import UIKit
+
+/// A custom UITextField that allows a label appearing near the clear button
+public class CustomTextField: UITextField {
+
+    private lazy var commonSpacing: CGFloat = 8
```
Comment (SergioEstevao) [inline Modules/Utils/Sources/PocketCastsUtils/Extensions/UI/CustomTextView.swift]: Maybe I may missing something but do you see this value as something that can be configured later on? Or could this be a constant?

**Comment by SergioEstevao** [comment]
From what I see in the Figma designs shouldn't the close button and search button be center aligned?

**Review by SergioEstevao — CHANGES_REQUESTED** [review]
General UI logic is working as you described.
I tested this on the iPhone and iPad multitask modes with multiple themes and it's looking great.
Just left some questions and nitpick comments to be addressed.

File: podcasts.xcodeproj/xcshareddata/xcschemes/pocketcasts.xcscheme (hunk 2 above)
Comment (leandroalonso) [reply podcasts.xcodeproj/xcshareddata/xcschemes/pocketcasts.xcscheme]: Ops, not! Reverted

File: podcasts/TranscriptSearchAccessoryView.swift (hunk 3 above)
Comment (leandroalonso) [reply podcasts/TranscriptSearchAccessoryView.swift:46]: Yes, I'll do it on the next PR! Just did that to test that the buttons are actually working as intended.

File: Modules/Utils/Sources/PocketCastsUtils/Extensions/UI/CustomTextView.swift (hunk 4 above)
Comment (leandroalonso) [reply Modules/Utils/Sources/PocketCastsUtils/Extensions/UI/CustomTextView.swift]: This is a constant!

I didn't want to create a static var though. I quite don't like the syntax on the view 🙈

It is a `var` because that's a requirement for `lazy`. But I can change to a `let` or something else. Any idea?

**Comment by leandroalonso** [comment]
> From what I see in the Figma designs shouldn't the close button and search button be center aligned?

Good point! Fixed! Ready for another look!

File: Modules/Utils/Sources/PocketCastsUtils/Extensions/UI/CustomTextView.swift (hunk 4 above)
Comment (SergioEstevao) [reply Modules/Utils/Sources/PocketCastsUtils/Extensions/UI/CustomTextView.swift]: I think a lazy var can still be changed after init, you just delaying the initialization to the first use.

For me it can be just a let, pretty sure the compiler will optimize that and just replace whenever it's used.

Comment (leandroalonso) [reply Modules/Utils/Sources/PocketCastsUtils/Extensions/UI/CustomTextView.swift]: Changed in 2df741f5721cb59e0ad06f116698501ba75cfcfd

**Review by SergioEstevao — APPROVED** [review]
Looking great! Thanks for addressing all suggestions!
:shipit:
